recipe/conda_build_config.yaml

```diff
@@ -0,0 +1,2 @@
+docker_image:                                   # [os.environ.get("BUILD_PLATFORM", "").startswith("linux-")]
+  - quay.io/condaforge/linux-anvil-comp7        # [os.environ.get("BUILD_PLATFORM") == "linux-64"]
```

Comment:
Is this the only actual change? Also, why are the conditions between the two lines different?

Reply:
Ahh, so there were other lines, but they aren't here now. Nevermind, was just curious as to what was needed.

Reply:
Hmm...thought we had previously determined the CentOS 7 images were safe to use given the compilers are CentOS 6 based. Is that not the case? Should we be revisiting this assumption more generally?

Reply:
Yes, this is the only change. I lazily copied the lines I needed from the conda-forhe pinning feedstock which is why the selectors differ.

Reply:
We determined it was safe assuming the build system only picked up stuff from the sysroot we packaged with our compilers. It appears that numba's build picked up symbols outside of our sysroot.

Reply:
So this is hack around a bug/feature in the numba build system. At least that is my current theory.

Reply:
I think this is a one-time thing.

Reply:
It would be good to understand what is getting picked up from the system and why that is so we can fix it properly

We might want to think of better ways to test and catch these issues. It's likely this is not the only time (or package) where we run into this problem

Reply:
Right now I am trying to unblock an actual issue I am facing at runtime. I do not have time to debug. I suggest we open another issue and merge this PR. Otherwise, I am totally blocked from using numba past 0.54.

Reply:
No objection to including this in the short term. Could you please file the issue with more detail about what happened? CI takes a while here anyways

Reply:
Great! See #92